src/network/networkd-dhcp-static-lease.c

```diff
@@ -116,3 +116,124 @@ static int lease_new_static(Network *network, const char *filename, unsigned sec
 
         return 0;
 }
+
+void network_drop_invalid_static_leases(Network *network) {
+        DHCPStaticLease *static_lease;
+
+        assert(network);
+
+        ORDERED_HASHMAP_FOREACH(static_lease, network->dhcp_static_leases_by_section)
+                if (dhcp_static_lease_section_verify(static_lease) < 0)
+                        dhcp_static_lease_free(static_lease);
+}
+
+
+int config_parse_dhcp_static_lease_address(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue, 
+                void *data,
+                void *userdata) {
+
+        Network *network;
+        _cleanup_(dhcp_static_lease_free_or_set_invalidp) DHCPStaticLease *current_lease = NULL;
+        union in_addr_union addr;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(userdata);
+
+        network = userdata;
+
+        r = lease_new_static(network, filename, section_line, &current_lease);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to create a new static lease: %s", rvalue);
+                return 0;
+        }
+
+        if (isempty(rvalue)) {
+                sd_dhcp_static_lease_unset_address(current_lease->static_lease);
+                return 0;
+        }
+
+        r = in_addr_from_string(AF_INET, rvalue, &addr);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r,
+                           "Failed to parse DHCPv4 IPv4 address data, ignoring static lease assignment: %s", rvalue);
+
+                return 0;
+        }
+
+        /* set address */
```

Comment:
this comment is not useful. the next line literally repeats the same words. No comments is better than redundant comments. Hence either add something useful here that isn't immediately visible from the code anyway, or drop the comment.